File: jscomp/test/gpr_2642_test.js
'use strict';


function isfree(id, _param) do
  while(true) do
    var param = _param;
    switch (param.tag | 0) do
      case --[ Pident ]--0 :
          return id == param[0];
      case --[ Pdot ]--1 :
          _param = param[0];
          continue ;
      case --[ Papply ]--2 :
          if (isfree(id, param[0])) do
            return true;
          end else do
            _param = param[1];
            continue ;
          end
      
    end
  end;
end

exports.isfree = isfree;
--[ No side effect ]--

File: jscomp/test/gpr_2642_test.lua
__console = {log = print};


function isfree(id, _param) do
  while(true) do
    param = _param;
    local ___conditional___=(param.tag | 0);
    do
       if ___conditional___ == 0--[[ Pident ]] then do
          return id == param[1]; end end 
       if ___conditional___ == 1--[[ Pdot ]] then do
          _param = param[1];
          ::continue:: ; end end 
       if ___conditional___ == 2--[[ Papply ]] then do
          if (isfree(id, param[1])) then do
            return true;
          end else do
            _param = param[2];
            ::continue:: ;
          end end  end end 
      
    end
  end;
end end

exports = {};
exports.isfree = isfree;
return exports;
--[[ No side effect ]]
